### PR TITLE
[WIP] Riemann SSL

### PR DESCRIPTION
**Pull request by sfriesel**

### CODE REVIEW

File: src/spootnik/reporter.clj [hunk 1]
```diff
@@ -55,29 +59,36 @@
...
+                   (if ssl
+                     (RiemannClient/wrap
+                       (doto (new TcpTransport host (int port))
+                         (-> .-sslContext
+                             (.set (SSL/sslContext
+                                     (:pkey ssl)
+                                     (:cert ssl)
+                                     (:authority ssl))))))
+                     (RiemannClient/tcp host (int port)))
+                   (RiemannClient/udp host (int port)))]
```
Comment (pyr) [inline src/spootnik/reporter.clj]: you're now defaulting to UDP which I think is a bad idea. you could change the above to (if (or (nil? protocol) (= protocol "tcp")) ....)

Comment (pyr) [inline src/spootnik/reporter.clj]: Come to think of it, you could switch to a cond, and consider a TLS protocol which would mean TCP + TLS.

Comment (sfriesel) [inline src/spootnik/reporter.clj]: the default is in line 74, "tcp"

Comment (pyr) [inline src/spootnik/reporter.clj]: Sorry, missed it.

Comment (sfriesel) [inline src/spootnik/reporter.clj]: > you could switch to a cond

you mean literally `protocol "ssl" :ssl {...}` or just reusing the `TcpTransport` creation for plain tcp too?

Comment (pyr) [inline src/spootnik/reporter.clj]: `{:protocol "tls" :tls {....}}`

This way you could:

```clojure
(case protocol
   "udp" (make-udp-client)
   "tls"   (make-tls-client)
           (make-tcp-client))

**Comment by pyr** [comment]
Thanks!

**Pull request merged**